src/vmm/src/memory_snapshot.rs

```diff
@@ -117,8 +118,11 @@ impl SnapshotMemory for GuestMemoryMmap {
         writer: &mut T,
         dirty_bitmap: &DirtyBitmap,
     ) -> std::result::Result<(), Error> {
-        let page_size = sysconf::page::pagesize();
         let mut writer_offset = 0;
+        let page_size = match unsafe { libc::sysconf(libc::_SC_PAGESIZE) } {
+            -1 => panic!("Failed to dump dirty pages: {}", errno::Error::last()),
```

Comment:
I think we need to return an error here instead of a panic.

Reply:
I chose to panic here instead of throwing an error because this is the behavior of `sysconf::page::pagesize()`:

```
fn priv_pagesize() -> usize {
    // sysconf(_SC_PAGESIZE) is required by POSIX 1003.1: http://www.unix.com/man-page/posix/3p/sysconf/
    ::raw::sysconf(::raw::SysconfVariable::ScPagesize).expect("sysconf(_SC_PAGESIZE) failed, but _SC_PAGESIZE is required by POSIX 1003.1") as usize
}
```

 and also because `libc::sysconf` can only return error when the parameter is not an accepted one (which would never be the case here). 

However, it would make for a more consistent and elegant approach to raise an error here. I will change the behavior.

Reply:
I've observed the same "panic" behaviour inside mmap.rs - line 79 - **enable_dirty_page_tracking**. I wonder if it is worth changing it there also to return an error? From my short investigation, I'd say we leave it like that, that since I believe will involve more complex changes, but it is worth taking it into consideration.

Reply:
We did take that into consideration at that time and decided it would be best to stick with the `panic` approach, because the vm-memory proxy uses errors imported from the upstream vm-memory crate.